server/src/main/java/org/elasticsearch/script/field/ScriptFieldSource.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.script.field;
+
+/** This interface is used to mark classes that generate fields for a script. */
+public interface ScriptFieldSource {
+
+    /** Returns the script field generated by this source. */
```

Comment:
This name is so close to the runtime fields naming, eg `IpFieldScript` that I think it's worth either renaming or being very specific.  Particularly because there is *also* [scripted fields](https://www.elastic.co/blog/using-painless-kibana-scripted-fields)

Reply:
Per discussion: rename to `ScriptFieldFactory` and cut an issue for renaming runtime fields.